plasmapy/atomic/names.py

```diff
@@ -454,6 +454,8 @@ def _extract_integer_charge(argument: str) -> Tuple[str, int]:
 
     """
 
+    warnings.warn("_extract_integer_charge is deprecated.", DeprecationWarning)
+
```

Comment:
Well... this is a good reason to make this PR less self-contained, I guess. What do you think is the best way to do this after these changes?

Reply:
My plan is to remove these private functions as part of this pull request since `parsing.py` and the `Particle` class should make these private functions no longer needed, so the DeprecationWarnings are primarily for my short-term benefit.  PyCharm also uses a strikethrough font to denote deprecated functions which makes them easier to find!

Reply:
It does? TIL, thanks! That's a good tactic, then. #stealingthisbuild